src/main/java/com/powsybl/openloadflow/equations/Equation.java

```diff
@@ -114,21 +114,14 @@ private static double getBusTargetV(LfBus bus) {
         if (bus.isDiscreteVoltageControlled()) {
             return bus.getDiscreteVoltageControl().getTargetValue();
         }
-        if (bus.getControllerBuses().isEmpty()) {
-            return bus.getTargetV();
-        } else {
-            List<LfBus> controllerBuses = bus.getControllerBuses()
-                    .stream()
-                    .filter(LfBus::hasVoltageControl)
-                    .collect(Collectors.toList());
-            if (bus.hasVoltageControl()) {
-                controllerBuses.add(bus);
-            }
-            if (controllerBuses.isEmpty()) {
-                throw new IllegalStateException("None of the controller buses of bus '" + bus.getId()
-                        + "'has voltage control on");
+        if (bus.getVoltageControl().isPresent() && bus.isVoltageControlled()) {
```

Comment:
I don't understand why you have the right to do that `(bus.getVoltageControl().isPresent() && bus.isVoltageControlled()` as the voltage control is optional. How you deal with that ?

Reply:
Indeed Sonar says that it is a bug :-)

Reply:
I corrected the sonar bug, I hope it is more clear now!